CMakeLists.txt

```diff
@@ -175,6 +179,9 @@ if(CMAKE_COMPILER_IS_GNU)
             set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wformat-signedness")
         endif()
     endif()
+    if (NOT CMAKE_C_STANDARD AND GCC_VERSION VERSION_LESS 5.0)
```

Comment:
What about to always set flag, whatever the compiler version. For later versions as it is their default it shouldn't matter and if at some point their default move to c11 we are still good.
Otherwise why not doing the same for the arm compiler?

Reply:
> For later versions as it is their default it shouldn't matter and if at some point their default move to c11 we are still good.

agree.

> Otherwise why not doing the same for the arm compiler?

if cmake version < 3.1, there is no portable approach to set -std=c99 for all compiler, we should do it by different compiler.
I add flags for gcc only currently.
BTY, what compiler should we support?

Reply:
> why not doing the same for the arm compiler?

I don't think Cmake has any support for Arm compilers. And it shouldn't try to invoke an Arm Compiler unless explicitly requested, because that could cause an unexpected network access to check the license. People who use a “specialist” compiler like Arm Compiler generally have their build environment set up to pass target-specific flags anyway, so Cmake defaults aren't particularly useful for them. The Cmake defaults are mostly useful for non-cross builds where the user just wants to build without fuss.